ruapu.h

```diff
@@ -334,6 +335,54 @@ RUAPU_INSTCODE(xtheadmempair, 0xe0a1450b) // th.lwd a0,a0,(sp),#0,3
 RUAPU_INSTCODE(xtheadsync, 0x0180000b) // th.sync
 RUAPU_INSTCODE(xtheadvdot, 0x8000600b) // th.vmaqa.vv v0,v0,v0
 
+// RVV 1.0 support
+static void ruapu_rvv_assert(int cond) {
+    // unimp
+    if (!cond)
+        asm volatile(".align 2\n.word 0xc0001073" : : : );
+}
+typedef intptr_t ruapu_riscv_xlen_t;
+// vcsr is only defined in rvv 1.0, which doesn't exist in rvv 0.7.1 or xtheadvector.
+static void ruapu_rvv1p0_avail() {
+    // csrr a0, vcsr
+    asm volatile(".word 0x00f02573");
+}
+__attribute__((naked))
+static ruapu_riscv_xlen_t ruapu_rvv_vsetvl(int vtype) {
+    // vsetvl a0, zero, a0
+    asm volatile(".word 0x80a07557\nret" : : : );
+}
+__attribute__((naked))
+static ruapu_riscv_xlen_t ruapu_rvv_vlenb() {
+    // csrr a0, vlenb
+    asm volatile(".word 0xc2202573\nret" : : : );
+}
+static ruapu_riscv_xlen_t ruapu_rvv_vsetvl_safe(int vtype) {
+    ruapu_riscv_xlen_t vl = ruapu_rvv_vsetvl(vtype);
+    // check vill bit
+    ruapu_rvv_assert(vl > 0);
+    return vl;
+}
```

Comment:
5 tiny functions are too fragmented.
Is there any way to merge them into 2, one for `RUAPU_DETECT_ZVL` and the other for `RUAPU_RVV_INSTCODE`?

Reply:
I can inline `ruapu_rvv_vsetvl_safe` into its callers.

Unfortunately, I don't know how to merge naked functions into their callers. I hardcode these instructions to make sure that ruapu can be compiled by old as without rvv-support.

Reply:
Fixed.